ext/node/polyfills/internal/crypto/diffiehellman.ts

```diff
@@ -1293,11 +1298,47 @@ export class ECDH {
   }
 }
 
-export function diffieHellman(_options: {
+const dhEnabledKeyTypes = new Set(["dh", "ec", "x448", "x25519"]);
+
+export function diffieHellman(options: {
   privateKey: KeyObject;
   publicKey: KeyObject;
 }): Buffer {
-  notImplemented("crypto.diffieHellman");
+  // @todo(iuioiua) This isn't calling the correct error class
+  validateObject(options, "options");
```

Comment:
For some reason, this throws with a generic `Error` class instance rather than `TypeError`.

Reply:
This seems to be the reason:
```ts
export class NodeTypeError extends NodeErrorAbstraction implements TypeError {
  constructor(code: string, message: string) {
    super(TypeError.prototype.name, code, message);
    Object.setPrototypeOf(this, TypeError.prototype);
    this.toString = function () {
      return `${this.name} [${this.code}]: ${this.message}`;
    };
  }
}
```

Reply:
How should we fix this? I'm sure this affects all use of `validateObject()` and other similar validation functions. IIUC, `NodeTypeError` should extend `TypeError` instead of `Error`. Perhaps, we could do something like:
```ts
export class NodeTypeError extends TypeError {
  code: string;

  constructor(name: string, code: string, message: string) {
    super(message);
    this.code = code;
    this.name = name;
    //This number changes depending on the name of this class
    //20 characters as of now
    this.stack = this.stack && `${name} [${this.code}]${this.stack.slice(20)}`;
  }

  override toString() {
    return `${this.name} [${this.code}]: ${this.message}`;
  }
}
```

Ref: https://github.com/denoland/deno/blob/c38f59f6579910b922d89b341d80e5fa01fee3d7/ext/node/polyfills/internal/errors.ts#L387

Reply:
I'm unsure why, but I'm no longer experiencing this issue 🥲